site/app/models/gradeable/Gradeable.php

```diff
@@ -471,6 +471,34 @@ private function parseDates(array $dates) {
         return $parsedDates;
     }
 
+   
+    public function setRandomPeerGradersList(&$input) {
```

Comment:
This should be generalized to `setPeerGradingList`. Error checking should still occur. On bad rows, do not set, and instead throw an exception.

Reply:
Because there is difference in the structure of input the functions take,  it was not generalized.